server/manager/src/main/java/org/apache/accumulo/manager/tableOps/delete/ReserveTablets.java

```diff
@@ -0,0 +1,110 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.manager.tableOps.delete;
+
+import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType.LOCATION;
+import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType.OPID;
+import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType.PREV_ROW;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiConsumer;
+
+import org.apache.accumulo.core.data.NamespaceId;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.dataImpl.KeyExtent;
+import org.apache.accumulo.core.fate.Repo;
+import org.apache.accumulo.core.metadata.schema.Ample;
+import org.apache.accumulo.core.metadata.schema.TabletOperationId;
+import org.apache.accumulo.core.metadata.schema.TabletOperationType;
+import org.apache.accumulo.manager.Manager;
+import org.apache.accumulo.manager.tableOps.ManagerRepo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReserveTablets extends ManagerRepo {
+
+  private static final Logger log = LoggerFactory.getLogger(ReserveTablets.class);
+
+  private static final long serialVersionUID = 1L;
+
+  private final TableId tableId;
+  private final NamespaceId namespaceId;
+
+  public ReserveTablets(TableId tableId, NamespaceId namespaceId) {
+    this.tableId = tableId;
+    this.namespaceId = namespaceId;
+  }
+
+  @Override
+  public long isReady(long tid, Manager manager) throws Exception {
+
+    var opid = TabletOperationId.from(TabletOperationType.DELETING, tid);
+
+    // The consumer may be called in another thread so use an AtomicLong
+    AtomicLong accepted = new AtomicLong(0);
+    BiConsumer<KeyExtent,Ample.ConditionalResult> resultsConsumer = (extent, result) -> {
+      if (result.getStatus() == Ample.ConditionalResult.Status.ACCEPTED) {
+        accepted.incrementAndGet();
+      } else {
+        log.debug("Failed to set operation id {} {}", opid, extent);
+      }
+    };
+
+    long locations = 0;
+    long otherOps = 0;
+    long submitted = 0;
+    long tabletsSeen = 0;
+
+    try (
+        var tablets = manager.getContext().getAmple().readTablets().forTable(tableId)
+            .fetch(OPID, PREV_ROW, LOCATION).checkConsistency().build();
+        var conditionalMutator =
+            manager.getContext().getAmple().conditionallyMutateTablets(resultsConsumer)) {
+      tabletsSeen++;
+      for (var tabletMeta : tablets) {
```

Comment:
This can sorta be done on the client side currently using streams. Can do the following with the current code, however the `tablets` object can not be used with the foreach loop later in the code because its a stream.

```java
 var tablets = manager.getContext().getAmple().readTablets().forTable(tableId)
            .fetch(OPID, PREV_ROW, LOCATION).checkConsistency().build().stream().filter(tm->!opid.equals(tm.getOperationId()));
```

One thing that may be nice is adding some sort of easy to use filters to ample that can push things down to server sdie iterators.  There is a uses case for this in the GC, I'll open an issue about exploring this further.

Reply:
That's a good point that we can just use the stream api, so that works so can just use that since it already exists. The sever side iterator would be a nice enhancement.

Reply:
Opened #3933